jazz_assets/components/crud/postSearch.js

```diff
@@ -0,0 +1,77 @@
+ /**
+      Search a Asset-Catalog entry in dynamodb table
+      @module: postSearch.js
+      @description: CRUD functions for asset catalog
+      @author:
+      @version: 1.0
+  **/
+
+ const utils = require("../utils.js"); //Import the utils module.
+ const logger = require("../logger.js")(); //Import the logging module.
+
+ module.exports = (query, asset_table, onComplete) => {
+     // initialize dynamodb
+     var docClient = utils.initDocClient();
+
+     var filter = "";
+     var insertAndString = " AND ";
+     var attributeValues = {};
+     var attributeNames = {};
+
+     var params = {
+         TableName: asset_table,
+         IndexName: global.global_config.ASSETS_DOMAIN_SERVICE_INDEX,
+         KeyConditionExpression: "#d = :service_domain and service = :service_name",
+         ExpressionAttributeValues: {
+             ":service_name": query.service,
+             ":service_domain": query.domain
+         },
+         ExpressionAttributeNames: {
+             "#d": "domain"
+         }
+     };
+
+     var keys_list = global.global_config.ASSET_SEARCH_OPTIONAL_FILTER_PARAMS;
+
+     // Generate filter string
+     keys_list.map((key) => {
+         var key_name = utils.getDatabaseKeyName(key);
+
+         if (query[key] && key_name) {
+             if (key_name === "type") {
```

Comment:
yes please remove and update to avoid any confusion

Reply:
updated